IndexHNSW.cpp

```diff
@@ -289,29 +289,30 @@ void IndexHNSW::search (idx_t n, const float *x, idx_t k,
 {
     FAISS_THROW_IF_NOT_MSG(storage,
        "Please use IndexHSNWFlat (or variants) instead of IndexHNSW directly");
-    size_t nreorder = 0;
+    HNSWStats stats;
 
     idx_t check_period = InterruptCallback::get_period_hint (
           hnsw.max_level * d * hnsw.efSearch);
 
     for (idx_t i0 = 0; i0 < n; i0 += check_period) {
         idx_t i1 = std::min(i0 + check_period, n);
 
-#pragma omp parallel reduction(+ : nreorder)
+#pragma omp parallel
         {
             VisitedTable vt (ntotal);
 
             DistanceComputer *dis = storage_distance_computer(storage);
             ScopeDeleter1<DistanceComputer> del(dis);
 
-#pragma omp for
+#pragma omp declare reduction (stats_combine : HNSWStats : omp_out.combine(omp_in))
```

Comment:
not supported for openmp < 4. 
If it is not too much work, I'd suggest doing it with reduction + 's

Reply:
This PR is just a sync, so we shouldn't introduce diverging changes here.